src/networking/client/plugin.rs

```diff
@@ -0,0 +1,56 @@
+use std::net::SocketAddr;
+
+use bevy::prelude::{AppBuilder, Plugin, IntoQuerySystem};
+use quinn::{crypto::rustls::TlsSession, generic::Connecting};
+use tokio::sync::mpsc::unbounded_channel;
+use url::Url;
+
+use crate::networking::{events::ReceiveEvent, events::SendEvent, systems::NetworkConnections, systems::SessionEventListenerState, systems::handle_connection, systems::receive_net_events, systems::send_net_events};
+
+pub struct Network {
+    pub addr: SocketAddr,
+    pub url: Url,
+    pub cert: quinn::CertificateChain
+}
+
+impl Plugin for Network {
+    fn build(&self, app: &mut AppBuilder) {
+        // Create mpsc endpoints for received network events and store them in a resources
+        let (send, recv) = unbounded_channel();
+        app.add_resource(SessionEventListenerState {
+            event_receiver: recv,
+            stream_senders: Default::default(),
+            send_event_reader: Default::default(),
+        });
+
+        app.add_resource::<NetworkConnections>(Default::default());
```

Comment:
I'm pretty sure this will be identical, and more idiomatic:
```rust
app.init_resource::<NetworkConnections>()
```
Bevy's `FromResources` is implemented automatically for everything implementing `Default`.